LinqToRest/QueryProvider.cs

```diff
@@ -34,15 +39,34 @@ public override object Execute(Expression expression)
             return Activator.CreateInstance(typeof(ProjectionReader<>).MakeGenericType(elementType), _resourceRetriever, uri);
         }
 
+        public TResult ExecuteAsync<TResult>(Expression expression, CancellationToken cancellationToken = default)
+        {
+            // Validate Generic Type is a Task (might need to be adjusted if we need to support ValueTasks as well)
+            if (!(typeof(TResult).GetGenericTypeDefinition() == typeof(Task<>)))
+            {
+                throw new ArgumentException("Type is expected to be a generic Task type.");
+            }
+
+            var result = Translate(expression);
+            var uri = new Uri(result.CommandText);
+            var elementType = TypeSystem.GetElementType(expression.Type);
+
+            var methodToExecute = typeof(ResourceRetriever).GetMethods()
+                .Single(method => method.Name == nameof(ResourceRetriever.RetrieveResource) && method.IsGenericMethod)
+                .MakeGenericMethod(elementType);
+
+            return (TResult) methodToExecute.Invoke(_resourceRetriever, new object[] {uri, cancellationToken});
+        }
+
         private TranslateResult Translate(Expression expression)
         {
             expression = Evaluator.PartialEval(expression);
             var queryBinder = _queryBinderFactory();
-            var proj = (ProjectionExpression)queryBinder.Bind(expression);
+            var proj = (ProjectionExpression) queryBinder.Bind(expression);
```

Comment:
Remove the space between type and value:
```suggestion
            var proj = (ProjectionExpression)queryBinder.Bind(expression);
```